src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -364,6 +352,61 @@ private void MoveCaretToClosingPoint()
                 }
             }
 
+            private void ApplyBraceCompletionResult(BraceCompletionResult result)
```

Comment:
is ths called from multiple places?  if not, should we move the transaction *into* this?

Reply:
this is called from multiple places.  In the first two instances it is used to insert the closing brace result, then apply the formatting changes result all in one transaction.  In the second instance it inserts the changes after return in a transaction.  So the transaction needs to be outside of this method.